### PR TITLE
Feature: add & remove 

### PR DESCRIPTION
**Pull request by memelopez**
this PR implements: 
- new JS file for update method
- new JS file for remove method
- new JS file for clearing completed tasks method
- all changes to the To Do List are saved in local storage

### CODE REVIEW

File: src/clearCompleted.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import returnsUncompleted from './isCompleted';
+
+export default function clearCompleted() {
+  // gets todos from local storage
+  const tasks = Store.getTasks();
+
+  // create new array from filter()
+  const uncompletedTasks = tasks.filter(returnsUncompleted);
+
+  Store.setTasks(uncompletedTasks);
```
Comment (sinansevgi) [inline src/clearCompleted.js:13]: - [ ] You should update indexes of remaining tasks after cleaning completed tasks. Right now it stays the same, as you can see in the screen recording below .

Please check [the project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#project-requirements)

Screencast-from-09-25-2021-01_39_49-AM

File: src/removeTask.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import Store from './store';
+
+export default function removeCompleted(index) {
+  // gets todos from local storage
+  const todos = Store.getTasks();
+
+  todos.splice(index, 1);
+
+  Store.setTasks(todos);
+}
```
Comment (sinansevgi) [inline src/removeTask.js:12]: - [x] You should update indexes of all tasks after removing a task.

File: src/index.js [hunk 3]
```diff
@@ -54,4 +54,45 @@ document.querySelector('#task-list').addEventListener('change', (e) => {
...
+    const newDesc = document.querySelector('#inputEdit').value;
+    UI.updateTask(index, newDesc);
+
+    // Reload page
+    location.reload();
+  }
+});
+
+document.querySelector('#pBtm').addEventListener('click', () => {
+  console.log('hi');
```
Comment (sinansevgi) [inline src/index.js]: - [x] Please remove all console outputs. Please check [the JavaScript best practices guide](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) for more information.

File: src/index.js [hunk 4]
```diff
@@ -54,4 +54,45 @@ document.querySelector('#task-list').addEventListener('change', (e) => {
...
+
+// Event: editing and removing
+document.querySelector('#task-list').addEventListener('click', (e) => {
+  const classesIcn = e.target.className;
+  const classesArr = classesIcn.split(' ');
+
+  const li = e.target.parentElement.parentElement;
+  const ulList = document.querySelector('#task-list');
+  const nodes = Array.from(ulList.children);
+  const index = nodes.indexOf(li);
```
Comment (sinansevgi) [inline src/index.js:67]: - [x] The index of the first to-do item should be `1` as request in [the project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#project-requirements). Currently it starts from `0`.

> By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).

File: src/index.js [hunk 5]
```diff
@@ -54,4 +54,45 @@ document.querySelector('#task-list').addEventListener('change', (e) => {
     // Reload page
     location.reload();
```
Comment (sinansevgi) [inline src/index.js:55]: [Optional]
- [ ] One of the main concepts that this project wants to improve is DOM manipulation. Nowadays pages are highly dynamic and responsive. Refreshing page after every action is not a good practice. You should be able to update a specific part of your page. This is not an issue that prevents the approval of your project but  I highly recommend you to remove all page refreshes and try to update specific parts of DOM when something changes.

**Review by sinansevgi — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
------

Hi @memelopez , 

Good job so far!
I really liked the focus effect and how the todo item changes to the input field when I try to edit a todo item :clap:
There are some issues that you still need to work on to go further but you are almost there!

 ## Required Changes ♻️
_Please check the inline comments for my suggestions_ :arrow_down:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 5 above)
Comment (memelopez) [reply src/index.js:55]: Thank you for the thorough review and the good advice 👍🏼 I'll try to implement the changes needed to remove the reloading of the page and I'll keep it in mind for future projects.

**Review by vichuge — APPROVED** [review]
# Approved project! ✅ 🥳 🎈 🎉🎉🎉🎉🎉🎉🎉🎉 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

InbornCandidHornet-size_restricted

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**